src/sentry/integrations/github_enterprise/integration.py

```diff
@@ -225,23 +225,29 @@ def build_integration(self, state):
             identity['access_token'],
             state['installation_id'])
 
+        domain_name = installation['account']['html_url'].replace('https://', '').split("/")[0]
+
         return {
             'name': installation['account']['login'],
-            'external_id': installation['id'],
+            # installation id is not enough to be unique for self-hosted GH
+            'external_id': '{}:{}'.format(domain_name, installation['id']),
             # GitHub identity is associated directly to the application, *not*
             # to the installation itself.
-            'idp_external_id': installation['app_id'],
+            # app id is not enough to be unique for self-hosted GH
+            'idp_external_id': '{}:{}'.format(domain_name, installation['app_id']),
             'metadata': {
                 # The access token will be populated upon API usage
                 'access_token': None,
                 'expires_at': None,
                 'icon': installation['account']['avatar_url'],
-                'domain_name': installation['account']['html_url'].replace('https://', '').split("/")[0],
+                'domain_name': domain_name,
+                'installation_id': installation['id'],
                 'installation': installation_data
             },
             'user_identity': {
                 'type': 'github_enterprise',
-                'external_id': user['id'],
+                # used id is not enough to be unique for self-hosted GH
+                'external_id': '{}:{}'.format(domain_name, user['id']),
```

Comment:
It is actually, since the identity is unique on (idp, external_id), and the `idp_external_id` is now the `domain_name:app_id`. No need for this

Reply:
That should have `idp__external_id='domain:app_id'`